src/Bowling.js

```diff
@@ -0,0 +1,43 @@
+function Bowling () {
+  this.rolls = []
+}
+Bowling.prototype.roll = function (pins) {
+  this.rolls.push(pins)
+}
+Bowling.prototype.score = function () {
+  var result = 0
+  var rollIndex = 0
+  var bowling = this
+  for (var frameIndex = 0; frameIndex < 10; frameIndex++) {
```

Comment:
There's a lot going on here. It might be worth separating out your concerns. Maybe have a class that creates frames and calculates that frame's score and then a class that manages the game.